homeassistant/components/seventeentrack/sensor.py

```diff
@@ -1,341 +1,160 @@
-        self.first_update = True
-
-    async def _async_update(self):
-        """Get updated data from 17track.net."""
-
-        try:
-            packages = await self._client.profile.packages(
-                show_archived=self._show_archived, tz=self.timezone
-            )
-            _LOGGER.debug("New package data received: %s", packages)
-
-            new_packages = {p.tracking_number: p for p in packages}
-
-            to_add = set(new_packages) - set(self.packages)
-
-            _LOGGER.debug("Will add new tracking numbers: %s", to_add)
-            if to_add:
-                self._async_add_entities(
-                    [
-                        SeventeenTrackPackageSensor(self, new_packages[tracking_number])
-                        for tracking_number in to_add
-                    ],
-                    True,
+    def extra_state_attributes(self):
+        """Return the device state attributes."""
+        package_data = {}
+        for package in self.coordinator.data["packages"]:
+            if package.status == self.sensor_type:
+                package_data.update(
+                    {
+                        package.tracking_number: {
+                            ATTR_FRIENDLY_NAME: package.friendly_name,
+                            ATTR_INFO_TEXT: package.info_text,
+                            ATTR_TIMESTAMP: package.timestamp,
+                            ATTR_STATUS: package.status,
+                            ATTR_LOCATION: package.location,
+                            ATTR_TRACKING_INFO_LANGUAGE: package.tracking_info_language,
+                            ATTR_PACKAGE_TYPE: package.package_type,
+                            ATTR_ORIGIN_COUNTRY: package.origin_country,
+                            ATTR_DESTINATION_COUNTRY: package.destination_country,
+                        }
+                    }
                 )
 
-            self.packages = new_packages
-        except SeventeenTrackError as err:
-            _LOGGER.error("There was an error retrieving packages: %s", err)
-
-        try:
-            self.summary = await self._client.profile.summary(
-                show_archived=self._show_archived
-            )
-            _LOGGER.debug("New summary data received: %s", self.summary)
-
-            # creating summary sensors on first update
-            if self.first_update:
-                self.first_update = False
-
-                self._async_add_entities(
-                    [
-                        SeventeenTrackSummarySensor(self, status, quantity)
-                        for status, quantity in self.summary.items()
-                    ],
-                    True,
-                )
+        if package_data:
+            self._attrs[ATTR_PACKAGES] = package_data
 
-        except SeventeenTrackError as err:
-            _LOGGER.error("There was an error retrieving the summary: %s", err)
-            self.summary = {}
+        return self._attrs
```

Comment:
We don't need to use an instance attribute here. Just return the dict we create from parsing the coordinator data.